src/bin/build_octree.rs

```diff
@@ -123,7 +142,7 @@ fn split_node<'a, 'b: 'a, P>(
     let (leaf_nodes, split_nodes): (Vec<_>, Vec<_>) =
         children
             .into_iter()
-            .partition(|n| n.num_points < MAX_POINTS_PER_NODE);
+            .partition(|n| !should_split_node(n, resolution));
```

Comment:
Would it make sense to invert this logic for readability, i.e. instead of `!should_split_node()` rather have `is_leaf_node()`?